mono/metadata/mono-debug.h

```diff
@@ -195,7 +195,11 @@ MonoDebugMethodAsyncInfo*
 mono_debug_lookup_method_async_debug_info (MonoMethod *method);
 
 MonoDebugSourceLocation *
-mono_debug_method_lookup_location (MonoDebugMethodInfo *minfo, int il_offset);
+mono_debug_method_lookup_location (MonoDebugMethodInfo *minfo, int il_offset)
+#ifdef MONO_LLVM_INTERNAL
```

Comment:
What is this ?

Reply:
mono-debug.h is public and mono-compiler.h is not.
MONO_LLVM_INTERNAL is defined in mono-compiler.h.
When I put it in unconditionally, it failed in the SDK build.

This is a workaround, that has a chance of not working, or having varying meaning, depending on the include order, or not being best option.


Maybe instead:
1. make mono-compiler.h public?
2. Move MONO_LLVM_INTERNAL to mono-public.h?
3. Move mono_debug_method_lookup_location to a non-public header and see if anything breaks?

Reply:
Maybe just mark it MONO_API_NO_EXTERN_C, independently of MONO_LLVM_LOADED.

Reply:
I went with MONO_API_NO_EXTERN_C and comments.
Moving MONO_LLVM_INTERNAL would mean worrying about config.h:
`#if !defined(_MSC_VER) && !defined(HOST_SOLARIS) && !defined(_WIN32) && !defined(__CYGWIN__) && !defined(MONOTOUCH) && HAVE_VISIBILITY_HIDDEN`